applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_nodal_displacement_response_function.cpp

```diff
@@ -182,7 +182,9 @@ namespace Kratos
         const VariableComponentType& r_traced_dof =
             KratosComponents<VariableComponentType>::Get(mTracedDofLabel);
 
-        return mpTracedNode->FastGetSolutionStepValue(r_traced_dof, 0);
+        const auto& r_primal_traced_node = rModelPart.GetNode(mpTracedNode->Id());
+
+        return r_primal_traced_node.FastGetSolutionStepValue(r_traced_dof, 0);
```

Comment:
```suggestion
        return rModelPart.GetNode(mpTracedNode->Id()).FastGetSolutionStepValue(r_traced_dof, 0);
```
then you don't need the temp-var
A little less clean though, therefore just a suggestion :)